src/cargo/core/compiler/timings.rs

```diff
@@ -218,17 +237,6 @@ impl<'cfg> Timings<'cfg> {
         unit_time
             .unlocked_units
             .extend(unlocked.iter().cloned().cloned());
-        if self.report_json {
-            let msg = machine_message::TimingInfo {
-                package_id: unit_time.unit.pkg.package_id(),
-                target: &unit_time.unit.target,
-                mode: unit_time.unit.mode,
-                duration: unit_time.duration,
-                rmeta_time: unit_time.rmeta_time,
-            }
-            .to_json_string();
-            crate::drop_println!(self.config, "{}", msg);
-        }
         self.unit_times.push(unit_time);
```

Comment:
There are some things that raise some flags for me
- This moves away from serializing `machine_message` to internal data structures.  Using code in `machine_message` helps raise awareness of the compatibility requirements when making changes and is more discoverable for people who want to parse the output
- Generally we print messages as we go rather than doing one bug final message
- The PR doesn't explain why some information was left out of the report for stablization

Overall, I'd recommend
- Find out the thought process for what data got stablized
- Create an issue for discussing stabilizing more information (Issues are much better for discussing requirements and design while PRs are much focusing on an implementation to resolve an Issue)
- Focus on incrementally printing `machine_message` structs.

Reply:
I can open an issue to discuss if you'd prefer, but I have a few question:
* The HTML reporting uses internal data structures exclusively.   The intent here was to emulate the HTML reporting. Are you suggesting that all of the reporting structs move to `machine_message`?
* Capturing stdout output printed on the fly in a meaningful way is a bit cumbersome.   Again, emulating the HTML reporting was a goal, so the larger final output at the end was used.  Should we do both?

Reply:
> Capturing stdout output printed on the fly in a meaningful way is a bit cumbersome. Again, emulating the HTML reporting was a goal, so the larger final output at the end was used. Should we do both?

Pretty much all cargo messages (except `cargo metadata`) communicate like this.  libtet also works this way with providing suite summary messages in addition to the individual tests.

> The HTML reporting uses internal data structures exclusively. The intent here was to emulate the HTML reporting. Are you suggesting that all of the reporting structs move to machine_message?

I can definitely understand this.  I love the idea of a separating out the view and model like this.  At times though, shortcuts are made because not everything is locked down yet.  This is why I was recommending looking into and having a discussion on why not everything is exposed.